### PR TITLE
Helper methods( Filtering and Updating) for Notification model

### LINKED ISSUE #182: Add helper filtering methods to Notification
We need helper methods that facilitate using the new Notification model. Currently we can only get and filter by actor, but other important methods include get and filter by receiver in order to display in inbox and also filter by unread, read, deleted and mark multiple notifications as deleted, undeleted, read, unread. Another helpful method is to filter by type of notification.

### PR DESCRIPTION
**Pull request by omarelkashef**
This PR adds helper methods  that facilitate provide important operations to the Notification model. That is, we can now get and filter by receiver which would be very useful for the inbox. Also, we can filter by unread, read, deleted, and the type of the notification and mark multiple notifications as deleted, undeleted, read, unread.

### CODE REVIEW

File: src/chigame/users/models.py [hunk 1]
```diff
@@ -93,24 +93,68 @@ class GroupInvitation(models.Model):
...
+        self.update(read=True)
+
+    def mark_all_deleted(self):
+        self.update(visible=False)
+
+    def restore_all_deleted(self):
+        self.update(visible=True)
+
+    def is_read(self):
+        return self.filter(read=True)
```
Comment (AndrewM131) [inline src/chigame/users/models.py:147]: My only question is when is_read, is_unread, is_deleted, and is_not_deleted, how does the system know to only mark a specific notification as read, unread, deleted or not deleted? Does it need an additional input?

**Review by AndrewM131 — CHANGES_REQUESTED** [review]

File: src/chigame/users/models.py (hunk 1 above)
Comment (omarelkashef) [reply src/chigame/users/models.py:147]: These methods are called on querysets so for example for is_read, the queryset will have Notification where read is True

Comment (AndrewM131) [reply src/chigame/users/models.py:147]: And does the filter individualize it so only one is affected, and not all?

Comment (omarelkashef) [reply src/chigame/users/models.py:147]: All Notification that fit this criteria are included in the queryset.

Comment (AndrewM131) [reply src/chigame/users/models.py:147]: I see, thank you for the clarification. the helper methods look properly implemented in the Notification model.

**Review by AndrewM131 — APPROVED** [review]
LGTM, thank you for clarifying my questions.

**Review by elizabethli31 — APPROVED** [review]
Tested all methods, and everything works great. Very complete helpers as always. Good work! Update from dev and I'll merge

**Comment by omarelkashef** [comment]
Closing statment:  In this pull request, we introduced helpful methods that streamline essential operations for the Notification model. With these additions, we can now retrieve and filter notifications by receiver. Additionally, we enabled filtering by unread, read, deleted, and notification type the ability as well as marking multiple notifications as deleted, undeleted, read, or unread. 

@24raniwalar worked on the mark_x() and is_x() methods
@omarelkashef  worked on the filter and get methods

**Comment by elizabethli31** [comment]
Issue Score: Excellent

Comments:
Nice set of functions, and really clean code. Great job!

**Pull request merged**